File: SDKs/googleapis.com/recaptchaenterprise/v1/java/lib/src/main/java/openapisdk/models/shared/GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification
 * Private password leak verification info.
**/
public class GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("encryptedLeakMatchPrefixes")
    public String[] encryptedLeakMatchPrefixes;
    public GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification withEncryptedLeakMatchPrefixes(String[] encryptedLeakMatchPrefixes) {
        this.encryptedLeakMatchPrefixes = encryptedLeakMatchPrefixes;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("encryptedUserCredentialsHash")
    public String encryptedUserCredentialsHash;
    public GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification withEncryptedUserCredentialsHash(String encryptedUserCredentialsHash) {
        this.encryptedUserCredentialsHash = encryptedUserCredentialsHash;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("lookupHashPrefix")
    public String lookupHashPrefix;
    public GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification withLookupHashPrefix(String lookupHashPrefix) {
        this.lookupHashPrefix = lookupHashPrefix;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("reencryptedUserCredentialsHash")
    public String reencryptedUserCredentialsHash;
    public GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification withReencryptedUserCredentialsHash(String reencryptedUserCredentialsHash) {
        this.reencryptedUserCredentialsHash = reencryptedUserCredentialsHash;
        return this;
    }
}

File: SDKs/googleapis.com/recaptchaenterprise/v1/python/src/sdk/models/shared/googlecloudrecaptchaenterprisev1privatepasswordleakverification.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerificationInput:
    r"""GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerificationInput
    Private password leak verification info.
    """
    
    encrypted_user_credentials_hash: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('encryptedUserCredentialsHash') }})
    lookup_hash_prefix: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lookupHashPrefix') }})
    

@dataclass_json
@dataclasses.dataclass
class GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification:
    r"""GoogleCloudRecaptchaenterpriseV1PrivatePasswordLeakVerification
    Private password leak verification info.
    """
    
    encrypted_leak_match_prefixes: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('encryptedLeakMatchPrefixes') }})
    encrypted_user_credentials_hash: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('encryptedUserCredentialsHash') }})
    lookup_hash_prefix: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lookupHashPrefix') }})
    reencrypted_user_credentials_hash: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('reencryptedUserCredentialsHash') }})
    
